sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/SumSqlAggregator.java

```diff
@@ -35,11 +36,17 @@
 
 public class SumSqlAggregator extends SimpleSqlAggregator
 {
+  /**
+   * Workaround to avoid unconditional rewrite of SUM into COUNT+SUM0.
```

Comment:
```suggestion
   * We use this custom aggregation function instead of builtin SqlStdOperatorTable.SUM to avoid transformation to COUNT+SUM0. See CALCITE-6020 for more details. It can be handled differently after CALCITE-6020 is fixed.
```

Reply:
btw following comments on CALCITE-6020, we would still have been able to plan it but it won't be the plan that drill test case expects. But then drill uses calcite too 🤔

Reply:
I might have been misleading: Druid is able to plan and execute the rewritten `COUNT` / `SUM0` things; but because `SUM0` is essentially a plain `SUM` there is no good reason to do that.

Reply:
updated the comment :+1: